admin-dev/themes/new-theme/js/pages/product/components/dropzone/DropzonePhotoSwipe.vue

```diff
@@ -0,0 +1,162 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *-->
+<template>
+  <div
+    class="pswp"
+    tabindex="-1"
+    role="dialog"
+    aria-hidden="true"
+  >
+    <div class="pswp__bg" />
+
+    <div class="pswp__scroll-wrap">
+      <div class="pswp__container">
+        <div class="pswp__item" />
+        <div class="pswp__item" />
+        <div class="pswp__item" />
+      </div>
+
+      <div class="pswp__ui pswp__ui--hidden">
+        <div class="pswp__top-bar">
+          <div class="pswp__counter" />
+
+          <button
+            type="button"
+            class="pswp__button pswp__button--close"
+            :title="$t('window.closePhotoSwipe')"
+          />
+
+          <button
+            type="button"
+            class="pswp__button pswp__button--share"
+            :title="$t('window.download')"
+          />
+
+          <button
+            type="button"
+            class="pswp__button pswp__button--fs"
+            :title="$t('window.toggleFullscreen')"
+          />
+
+          <button
+            type="button"
+            class="pswp__button pswp__button--zoom"
+            :title="$t('window.zoomPhotoSwipe')"
+          />
+
+          <div class="pswp__preloader">
+            <div class="pswp__preloader__icn">
+              <div class="pswp__preloader__cut">
+                <div class="pswp__preloader__donut" />
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div
+          class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap"
+        >
+          <div class="pswp__share-tooltip" />
+        </div>
+
+        <button
+          type="button"
+          class="pswp__button pswp__button--arrow--left"
+          :title="$t('window.previousPhotoSwipe')"
+        />
+
+        <button
+          type="button"
+          class="pswp__button pswp__button--arrow--right"
+          :title="$t('window.nextPhotoSwipe')"
+        />
+
+        <div class="pswp__caption">
+          <div class="pswp__caption__center" />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+  import PhotoSwipe from 'photoswipe';
+  // eslint-disable-next-line
+  import PhotoSwipeUI_Default from "photoswipe/dist/photoswipe-ui-default";
+  import ProductMap from '@pages/product/product-map';
+  import ProductEventMap from '@pages/product/product-event-map';
+
+  const PhotoSwipeMap = ProductMap.dropzone.photoswipe;
+  const PhotoSwipeEventMap = ProductEventMap.dropzone.photoswipe;
+
+  export default {
+    name: 'DropzonePhotoSwipe',
+    props: {
+      files: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    mounted() {
+      const pswpElement = document.querySelectorAll(PhotoSwipeMap.element)[0];
```

Comment:
Shouldn't we check if element exists?

Reply:
No need, because this components is not used if no images are inside the page

Reply:
Besides since you only get the first element you could use `querySelector` only